test/rewards/AutoCompoundingRewards.ts

```diff
@@ -197,15 +197,32 @@ describe('AutoCompoundingRewards', () => {
         const EXP_DECAY_HALF_LIFE = duration.days(560);
         const EXP_DECAY_MAX_DURATION = EXP2_INPUT_TOO_HIGH.mul(EXP_DECAY_HALF_LIFE).sub(1).ceil().toNumber();
 
-        const END_TIME = (distributionType: RewardsDistributionType) =>
+        const programEndTime = (distributionType: RewardsDistributionType) =>
             distributionType === RewardsDistributionType.Flat ? START_TIME + FLAT_TOTAL_DURATION : 0;
 
-        const HALF_LIFE = (distributionType: RewardsDistributionType) =>
-            distributionType === RewardsDistributionType.Flat ? 0 : EXP_DECAY_HALF_LIFE / duration.days(1);
+        const programHalfLife = (distributionType: RewardsDistributionType) =>
+            distributionType === RewardsDistributionType.Flat ? 0 : EXP_DECAY_HALF_LIFE;
 
-        const DURATION = (distributionType: RewardsDistributionType) =>
+        const programDuration = (distributionType: RewardsDistributionType) =>
             distributionType === RewardsDistributionType.Flat ? FLAT_TOTAL_DURATION : EXP_DECAY_MAX_DURATION;
 
+        const createProgram = async (
```

Comment:
I'd add `createFlatProgram` and `createExpProgram` here as well so that the tests/profiling is clearer as well

Reply:
Are you sure?
That would "mess up" (increase in size) all the tests which use this function.
The idea was to keep them "clean" and handle this "duality" in a single place.

Reply:
It's a good tradeoff. Although it'll increase the code, it'll be always 100% clear how each program is set.

Reply:
But it'll just move the `distributionType === RewardsDistributionType.Flat ?` check from this function into each and every one of the tests which use it.

Reply:
It will also abstract away distribution-specific inputs (like in the contracts themselves), I believe

Reply:
can also just separate the tests and not using `if` in most cases (although I can see how that will be annoying with general management functions).
another approach would be to create separate functions + another function that receives a `type` arg to be used in the management tests, while calling the dedicated functions directly in the financial tests.
but in general I agree that having a function with all permutations is less readable.

Reply:
The latter is how I imagined it as well

Reply:
done